pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -64,8 +66,8 @@ func (pl *InterPodAffinity) EventsToRegister() []framework.ClusterEventWithHint
 		// an unschedulable Pod schedulable.
 		// - Add. An unschedulable Pod may fail due to violating pod-affinity constraints,
 		// adding an assigned Pod may make it schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}},
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeLabel}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}, QueueingHintFn: pl.isSchedulableAfterPodChange},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
```

Comment:
memo for later: Depending on the decision made in https://github.com/kubernetes/kubernetes/pull/122292, we don't have to register framework.Update, but framework.UpdateNodeTaint. 
@nayihz You don't have to change here for now since #122292 hasn't reached the final decision, but may have to later. I'll re-visit this comment when this PR gets `/lgtm` and `/approve`.

/hold for confirm this before merging this PR.